packages/@webex/plugin-meetings/test/unit/spec/stats-analyzer/index.js

```diff
@@ -1718,6 +1732,41 @@ describe('plugin-meetings', () => {
           }
         ]);
       });
-    });
+      it('sends streams according to their is requested flag', async () => {
+        it('should send a stream if it is requested', async () => {
+          await startStatsAnalyzer();
+          fakeStats.audio.senders[0].report[0].isRequested = true;
+          await progressTime(MQA_INTERVAL);
+          assert.strictEqual(mqeData.audioTransmit[0].streams.length, 1);
+        });
+
+        it('should not sent a stream if its is requested flag is undefined', async () => {
+          fakeStats.audio.senders[0].report[0].isRequested = undefined;
+          await progressTime(MQA_INTERVAL);
+          assert.strictEqual(mqeData.audioTransmit[0].streams.length, 0);
+        });
+
+        it('should not send a stream if it is not requested', async () => {
+          fakeStats.audio.receivers[0].report[0].isRequested = false;
+          await progressTime(MQA_INTERVAL);
+          assert.strictEqual(mqeData.audioReceive[0].streams.length, 0);
+        });
+
+        it('should send a stream if it was requested in the last 60 seconds', async () => {
+          fakeStats.audio.receivers[0].report[0].lastRequestedUpdateTimestamp = performance.timeOrigin + performance.now() - 30 * 1000;
+          fakeStats.audio.receivers[0].report[0].isRequested = false;
+          await progressTime(MQA_INTERVAL);
+          assert.strictEqual(mqeData.audioReceive[0].streams.length, 1);
+        })
+        it('should not send a stream if it was not requested in the last 60 seconds', async () => {
+          fakeStats.audio.receivers[0].report[0].lastRequestedUpdateTimestamp = performance.timeOrigin + performance.now() + 30 * 1000;
```

Comment:
To achieve consistent results in tests, especially with time values, it's better to use stubs to consistently return expected results.

like

```
const stub = sinon.stub(performance, 'now');
stub.returns(60000);
```

Reply:
Please correct me if I am wrong, but is this necessary? 
We already have: 
```javascript
const progressTime = async (time = initialConfig.analyzerInterval) => {
await clock.tickAsync(time);
await testUtils.flushPromises();
};
```

Given the `progressTime` function we have, it looks like we're already using [sinon's fake timers](https://github.com/webex/webex-js-sdk/blob/2ea5778de5128968082a07a61b320d0dd3103417/packages/%40webex/plugin-meetings/test/unit/spec/stats-analyzer/index.js#L306) to simulate the passage of time in our tests. The `clock.tickAsync(time)` function advances the clock by a specified amount of time and resolves any promises that are due to be resolved after this simulated time has passed.

Reply:
When writing tests, mocking time helps ensure that you get the same expected result each time you run the test. The goal is to stub unreliable functions and write assertions based on the expected behavior of your code when a specific value is returned by performing.now(). This approach makes your tests more predictable and reliable.

Reply:
Done